### ISSUE TITLE
strictEqual / deepStrictEqual behave differently on numbers

### ISSUE THREAD

**Issue by rotu** [chaijs/chai#1532]
Chai's behavior of `strictEqual` and `deepStrictEqual` differ in how they treat numbers (particularly `NaN` and `-0`). It seems like "depth" should not affect the equality of two shallow objects.

```
chai.assert.strictEqual(NaN,NaN) // fails
chai.assert.deepStrictEqual(NaN,NaN) // passes

chai.assert.strictEqual(-0,0) // passes
chai.assert.deepStrictEqual(-0,0) // fails
```

Note that the documentation adds confusion, since it says that "deep equality" refers to `deep-eql` semantics and "strict equality" refers to `===`:
https://github.com/chaijs/chai/blob/1a8247f30dbe0f54268a9748ae673caec75d6bfe/lib/chai/core/assertions.js#L1716-L1718

It appears `deepStrictEqual` was originally added as an alias for `strictEqual` to match Node: https://github.com/chaijs/chai/pull/653

**Comment by koddsson** [comment]
I guess `NaN !== NaN` so it does make some sort of sense.

**Comment by koddsson** [comment]
Also `-0 === 0` now that I think about it.

**Comment by koddsson** [comment]
Then reading the `deep-eql` README, it says:

>- Strict equality for non-traversable nodes according to [Object.is](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is):
>. - `eql(NaN, NaN).should.be.true;`
>. - `eql(-0, +0).should.be.false;`

**Comment by koddsson** [comment]
Makes me think this is indeed working as expected but maybe a bit confusing.

**Comment by rotu** [comment]
> Makes me think this is indeed working as expected but maybe a bit confusing.

I don't know what you mean by "as expected". The discrepancy was not the result of a typo, but it's also not a reasonable design choice.

I think I would expect chai to change [`strictEqual` as `node:assert` did in v10](https://nodejs.org/api/assert.html#assertstrictequalactual-expected-message). See [relevant Node PR](https://github.com/nodejs/node/pull/17003).

**Comment by ahrjarrett** [comment]
I agree with @koddsson that `strictEqual` is technically working as expected. But like @rotu, I was also surprised by this behavior.

Perhaps the real issue is with `deepStrictEqual`. These days, [strict equality is a precise term](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-isstrictlyequal), and as far as I can tell, `deepStrictEqual` does not use `"==="` anywhere in its implementation.

Could we update the [docs](https://github.com/chaijs/chai/blob/97b218e1346cb2d3a84fbacbbf24d82c08cb0450/lib/chai/interface/assert.js#L203), to tell users that "deeply equal" means that leaves/primitives will be compared using [SameValue](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-samevalue) equality?
